pkg/platform/model/buildplanner/build.go

```diff
@@ -74,12 +74,12 @@ func (b *BuildPlanner) FetchCommit(commitID strfmt.UUID, owner, project string,
 		return nil, errs.Wrap(err, "failed to unmarshal build plan")
 	}
 
-	expression, err := buildexpression.New(commit.Expression)
+	script, err := buildscript.UnmarshalBuildExpression(commit.Expression, ptr.To(time.Time(commit.AtTime)))
 	if err != nil {
-		return nil, errs.Wrap(err, "failed to parse build expression")
+		return nil, errs.Wrap(err, "failed to parse build script")
```

Comment:
nit: technically it's a failure parsing the incoming `commit.Expression`, which is a build expression, right?